src/app/clusters/basic/basic.cpp

```diff
@@ -83,26 +83,85 @@ class PlatformMgrDelegate : public DeviceLayer::PlatformManagerDelegate
 
 PlatformMgrDelegate gPlatformMgrDelegate;
 
-} // anonymous namespace
-
-void emberAfBasicClusterServerInitCallback(chip::EndpointId endpoint)
+class BasicAttrAccess : public app::AttributeAccessInterface
 {
-    EmberAfStatus status;
-
-    char nodeLabel[DeviceLayer::ConfigurationManager::kMaxNodeLabelLength + 1];
-    if (ConfigurationMgr().GetNodeLabel(nodeLabel, sizeof(nodeLabel)) == CHIP_NO_ERROR)
+public:
+    BasicAttrAccess() : app::AttributeAccessInterface(Optional<EndpointId>::Missing(), app::Clusters::Basic::Id) {}
+    CHIP_ERROR Read(const app::ConcreteReadAttributePath & aPath, app::AttributeValueEncoder & aEncoder) override
     {
-        status = Attributes::NodeLabel::Set(endpoint, chip::CharSpan(nodeLabel, strlen(nodeLabel)));
-        VerifyOrdo(EMBER_ZCL_STATUS_SUCCESS == status, ChipLogError(Zcl, "Error setting Node Label: 0x%02x", status));
+        if (aPath.mAttributeId == app::Clusters::Basic::Attributes::NodeLabel::Id)
+        {
+            char nodeLabel[DeviceLayer::ConfigurationManager::kMaxNodeLabelLength + 1];
+            if (ConfigurationMgr().GetNodeLabel(nodeLabel, sizeof(nodeLabel)) == CHIP_NO_ERROR)
+            {
+                CharSpan span(nodeLabel, strlen(nodeLabel));
+                aEncoder.Encode(span);
+            }
+        }
+        else if (aPath.mAttributeId == app::Clusters::Basic::Attributes::Location::Id)
+        {
+            char location[DeviceLayer::ConfigurationManager::kMaxLocationLength + 1];
+            size_t codeLen = 0;
+            if (ConfigurationMgr().GetCountryCode(location, sizeof(location), codeLen) == CHIP_NO_ERROR)
```

Comment:
Do you happen to know if this custom accessor will be called when an application uses `Basic::Attributes::Location::Get()` accessor? I used `Basic::Attributes::Location::Get` in `OTARequestor.cpp` to build a request based on what is put int the Location attribute and wonder if this change may break it.

Reply:
![image](https://user-images.githubusercontent.com/1230620/145659837-8d7c7a40-1a51-4a42-b425-7c3ebfb144e2.png)

As above image shows If a read command is from network, this function will be called at src/app/util/ember-compatibility-functions.cpp::findAttributeAccessOverride

But if you using Basic::Attributes::Location::Set(), the value will be saved in src/app/util/attribute-storage.cpp. This will be called from src/app/util/ember-compatibility-functions.cpp:: emAfReadOrWriteAttribute ( some lines after findAttributeAccessOverride)
![WX20211211-101248](https://user-images.githubusercontent.com/1230620/145660467-e32d4450-7a35-433c-aade-c0bb4d1c55ea.png)

So, if you want to read a cluster attr from local, I think ReadSingleClusterData should get the same result but I did not try it. While, for more easy cases, I think ConfigurationMgr() should be OK.

Reply:
@Damian-Nordic Accessors functions don't take into account overrides.

We should not be directly poking the attribute store to get state from one cluster in another cluster.  Where clusters store state is an implementation detail.  What we should have instead are APIs for getting the relevant state that the cluster implements and that consumers can use.